src/utilities/routeMatchRegexRules.spec.ts

```diff
@@ -0,0 +1,151 @@
+import { describe, expect, test } from 'vitest'
+import { Route } from '@/types'
+import { resolveRoutes, routePathMatches, routeQueryMatches } from '@/utilities'
+import { component } from '@/utilities/testHelpers'
+
+describe('routePathMatches', () => {
+  test.each([
+    ['we*23mf#0'],
+    ['www.kitbag.io/'],
+    ['http://www.kitbag.io'],
+    ['http://www.kitbag.io/'],
+    ['http://www.kitbag.io/is/empty'],
+    ['http://www.kitbag.io/is/empty?with=query'],
+  ])('given url and route.path that does NOT match, returns false', (url) => {
+    const route = {
+      name: 'not-matches',
+      path: '/not/empty',
+      component,
+    } as const satisfies Route
+    const [resolved] = resolveRoutes([route])
+
+    const response = routePathMatches(resolved, url)
+
+    expect(response).toBe(false)
+  })
+
+  test.each([
+    ['/without/params'],
+    ['www.kitbag.io/without/params'],
+    ['http://www.kitbag.io/without/params'],
+  ])('given url and route.path WITHOUT params that does match, returns true', (url) => {
+    const route = {
+      name: 'no-params',
+      path: '/without/params',
+      component,
+    } as const satisfies Route
+    const [resolved] = resolveRoutes([route])
+
+    const response = routePathMatches(resolved, url)
+
+    expect(response).toBe(true)
+  })
+
+  test.each([
+    ['/with/true/params/true'],
+    ['/with/%/params/%'],
+    ['www.kitbag.io/with/123/params/true'],
+    ['http://www.kitbag.io/with/a/params/b'],
+  ])('given url and route.path with params that does match, returns true', (url) => {
+    const route = {
+      name: 'no-params',
+      path: '/with/:some/params/:inPath',
+      component,
+    } as const satisfies Route
+    const [resolved] = resolveRoutes([route])
+
+    const response = routePathMatches(resolved, url)
+
+    expect(response).toBe(true)
+  })
+})
+
+describe('routeQueryMatches', () => {
+  test.each([
+    ['we*23mf#0'],
+    ['www.kitbag.io/'],
+    ['http://www.kitbag.io'],
+    ['http://www.kitbag.io/'],
+    ['http://www.kitbag.io/is/empty'],
+    ['http://www.kitbag.io/is/empty?with=query'],
+  ])('given url and route.query that does NOT match, returns false', (url) => {
+    const route = {
+      name: 'not-matches',
+      path: '',
+      query: 'not=empty',
+      component,
+    } as const satisfies Route
+    const [resolved] = resolveRoutes([route])
+
+    const response = routeQueryMatches(resolved, url)
+
+    expect(response).toBe(false)
+  })
+
+  test.each([
+    ['?without=params&static=true'],
+    ['www.kitbag.io?without=params&static=true'],
+    ['www.kitbag.io/some/path?without=params&static=true'],
```

Comment:
I wouldn't expect this case to return true. Because the route has a path of `''` but and this path wouldn't match right?

Reply:
you're right that if we were testing `routeMatch` this case would fail, but in this case we're only testing `routeQueryMatches`